shared/src/main/scala/govariants/Board.scala

```diff
@@ -134,6 +137,45 @@ class Board(val size: Int) {
     }
   }
 
+  def compute_groups_from_scratch() = {
+    val visited: HashMap[Intersection, Boolean] = HashMap()
```

Comment:
Should probably be a set of intersections instead of a map (set membership will be the boolean value we store here).